### ISSUE TITLE
Simulator should return warrior metadata

### ISSUE THREAD

**Issue by dougajmcdonald** [gareththegeek/corewar#66]
When I load simulators, the warrior doesn't return the metadata in the parse results from what I can see :(

**Comment by gareththegeek** [comment]
Did you specify metadata using the comments? e.g.
```
;name imp
;author traditional
``` 
And do you mean the parse result before it hits the simulator or the loaded warrior after it has been through the initialise function?

**Comment by dougajmcdonald** [comment]
I got the metadata from the warriors in the warriors folder. The metadata comes back from the parser in tokens but not in the warrior post init

**Comment by gareththegeek** [comment]
Yeah, not sure if I ever bothered hooking that up but shouldn't be a problem.

**Comment by gareththegeek** [comment]
So is the problem that the meta data is returned from the parser in the `IParseResult` but that it is not within the `IState` after the simulator has been initialised?

The code certainly seems to be there to do this (see WarriorLoader.ts line 43).

I don't understand your description because you say that the metadata is in the result from the parser but not in parseResults - but the parse results are produced by the parser...?

**Comment by dougajmcdonald** [comment]
Sorry what I mean is that it's returned in the `IParseResult` within the tokens, but not within the `metaData` object, not the metaData object anywhere else.

So if I parse this:

```
;redcode
;name Dwarf
;author A. K. Dewdney
;assert CORESIZE % 5 == 0

	DAT		-1
start	ADD	#5,	-1
	MOV	#0,	@-2
	JMP	-2

	END	start
```

I get this:

![image](https://user-images.githubusercontent.com/1190042/33456254-bf84b8a0-d616-11e7-8d33-d9a68505b2dd.png)

Where `warrior` is the tokens run through the serialiser

**Comment by gareththegeek** [comment]
Ah, ok interesting.  I'll take a look!

**Comment by gareththegeek** [comment]
Should be resolved by #73

**Comment by dougajmcdonald** [comment]
Close 'em off sucker!

![image](https://user-images.githubusercontent.com/1190042/33458893-e94ae232-d61f-11e7-9e68-1158b19b1449.png)
